#! /bin/sh /usr/share/dpatch/dpatch-run
## 90_leap_year_bug.dpatch by Bernd Zeimetz <bzed@debian.org>
##
## DP:  Leap year bugfix, make gpsd work again.

@DPATCH@
diff -urNad gpsd-2.33~/gpsutils.c gpsd-2.33/gpsutils.c
--- gpsd-2.33~/gpsutils.c	2006-02-27 09:54:35.000000000 +0100
+++ gpsd-2.33/gpsutils.c	2008-01-23 16:31:48.000000000 +0100
@@ -82,6 +82,9 @@
     result += (year - 1968) / 4;
     result -= (year - 1900) / 100;
     result += (year - 1600) / 400;
+    if ((year % 4) == 0 && ((year % 100) != 0 || (year % 400) == 0) &&
+	(t->tm_mon % MONTHSPERYEAR) < 2)
+	     result--;
     result += t->tm_mday - 1;
     result *= 24;
     result += t->tm_hour;
